libs/langchain-community/src/llms/ibm.ts

```diff
@@ -0,0 +1,567 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import { CallbackManagerForLLMRun } from "@langchain/core/callbacks/manager";
+import { BaseLLM, BaseLLMParams } from "@langchain/core/language_models/llms";
+import { WatsonXAI } from "@ibm-cloud/watsonx-ai";
+import {
+  DeploymentsTextGenerationParams,
+  DeploymentsTextGenerationStreamParams,
+  DeploymentTextGenProperties,
+  ReturnOptionProperties,
+  TextGenerationParams,
+  TextGenerationStreamParams,
+  TextGenLengthPenalty,
+  TextGenParameters,
+  TextTokenizationParams,
+  TextTokenizeParameters,
+} from "@ibm-cloud/watsonx-ai/dist/watsonx-ai-ml/vml_v1.js";
+import {
+  Generation,
+  LLMResult,
+  GenerationChunk,
+} from "@langchain/core/outputs";
+import { BaseLanguageModelCallOptions } from "@langchain/core/language_models/base";
+import { AsyncCaller } from "@langchain/core/utils/async_caller";
+import { authenticateAndSetInstance } from "../utils/ibm.js";
+import {
+  GenerationInfo,
+  ResponseChunk,
+  TokenUsage,
+  WatsonxAuth,
+  WatsonxParams,
+} from "../types/watsonx_ai.js";
+
+/**
+ * Input to LLM class.
+ */
+
+export interface WatsonxCallOptionsLLM
+  extends BaseLanguageModelCallOptions,
+    Omit<
+      Partial<
+        TextGenerationParams &
+          TextGenerationStreamParams &
+          DeploymentsTextGenerationParams &
+          DeploymentsTextGenerationStreamParams
+      >,
+      "input"
+    > {
+  maxRetries?: number;
+}
+
+export interface WatsonxInputLLM
+  extends TextGenParameters,
+    WatsonxParams,
+    BaseLLMParams {
+  streaming?: boolean;
+}
+
+/**
+ * Integration with an LLM.
+ */
+export class WatsonxLLM<
+    CallOptions extends WatsonxCallOptionsLLM = WatsonxCallOptionsLLM
+  >
+  extends BaseLLM<CallOptions>
+  implements WatsonxInputLLM
+{
+  // Used for tracing, replace with the same name as your class
+  static lc_name() {
+    return "Watsonx";
+  }
+
+  lc_serializable = true;
+
+  streaming = false;
+
+  modelId = "ibm/granite-13b-chat-v2";
```

Comment:
We've standardized this param as `model`

Also would suggest not providing a default since it will become outdated